tools/corda-runtime-gradle-plugin/src/main/kotlin/net/corda/gradle/plugin/cordapp/CordappTasksImpl.kt

```diff
@@ -0,0 +1,194 @@
+package net.corda.gradle.plugin.cordapp
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import net.corda.gradle.plugin.configuration.ProjectContext
+import net.corda.gradle.plugin.dtos.GroupPolicyDTO
+import net.corda.gradle.plugin.exception.CordaRuntimeGradlePluginException
+import java.io.File
+import java.io.FileInputStream
+
+class CordappTasksImpl(var pc: ProjectContext){
+
+    /**
+     * Creates a group Policy based on the network config file and outputs it to the Group Policy json file.
+     */
+    fun createGroupPolicy() {
+        val groupPolicyFile = File(pc.groupPolicyFilePath)
+        val networkConfigFile = File("${pc.project.rootDir}${pc.networkConfig.configFilePath}")
+
+        val pluginsDir = "${pc.cordaCliBinDir}/plugins/"
+
+        if (!groupPolicyFile.exists() || groupPolicyFile.lastModified() < networkConfigFile.lastModified()) {
+
+            pc.logger.quiet("Creating the Group policy.")
+            val configX500Ids = pc.networkConfig.x500Names
+
+            GroupPolicyHelper().createStaticGroupPolicy(
+                groupPolicyFile,
+                configX500Ids,
+                pc.javaBinDir,
+                pluginsDir,
+                pc.cordaCliBinDir
+            )
+        } else {
+            pc.logger.quiet("Group policy up to date.")
+        }
+        validateGroupPolicy()
+    }
+
+    private fun validateGroupPolicy() {
+        val groupPolicyFile = File(pc.groupPolicyFilePath)
+        val groupPolicy = try {
+            val fis = FileInputStream(groupPolicyFile)
+            val mapper = ObjectMapper()
+            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            mapper.readValue(fis, GroupPolicyDTO::class.java)
+        } catch (e: Exception) {
+            val firstLine = groupPolicyFile.useLines { it.firstOrNull() }
+            if ((firstLine != null) && firstLine.contains("Unable to access jarfile \\S*corda-cli.jar".toRegex())) {
+                throw CordaRuntimeGradlePluginException("Unable to find the Corda CLI, has it been installed?")
+            }
+
+            throw CordaRuntimeGradlePluginException("Failed to read GroupPolicy from group policy file with exception: $e.")
+        }
+        validateGroupPolicyHasAllMembers(groupPolicy)
+    }
+
+    private fun validateGroupPolicyHasAllMembers(groupPolicy: GroupPolicyDTO){
+        val membersNames = groupPolicy.protocolParameters?.staticNetwork?.members?.map { it.name }
+        pc.networkConfig.vNodes.forEach { requiredNode ->
+            if (membersNames?.any { requiredNode.x500Name == it } != true) {
```

Comment:
I'm not looking at this in an IDE but is is this not just:
```suggestion
            if (!membersNames?.contains(requiredNode.x500Name)) {
```